### DISCUSSION TITLE
Event Handlers Not Firing - InMemory and Azure Service Bus Transport Method

### DISCUSSION THREAD

**Discussion by alsahe26** [rebus-org/Rebus#1144] [Q&A]
Hi all!

I have a simple pub (Web.API) /sub (Console) implementation with a very strange behaviour. With Rabbit.MQ as transport,  the event handlers fires correcty but when i change to inMemory or Azure Service Bus transport, didn't work. Some recommendation?

.NET 8 with the latest rebus packages (8.2.2).

Pub (Web.API):

![image](https://github.com/rebus-org/Rebus/assets/5134473/1bd1a89f-d399-42ed-9b47-631a7f5093ae)

Sub (Console):

![image](https://github.com/rebus-org/Rebus/assets/5134473/5e799f70-9335-4a47-ae01-6d5804f0bf4f)

Any help? Thank you!

**Comment by mookid8000** [comment]
Your code looks fine, but I do have one question: What do you mean when you say that your "event handlers aren't firing"? 

1: How do you publish the events?
2: Where do you subscribe to them?
3: How do you determine whether an event gets handled or not?

Please provide me with more details and I'll help you solve your problem 🙂

**Reply by alsahe26** [reply]
Hi Mogens! Thank you for you response!

**What do you mean when you say that your "event handlers aren't firing"?**
I try to explain that the message that I publish in the pub, doesn't arrive to Event Handler (with RabbitMQ works!). The example solution runs two projects WebAPI - Pub and Console - Sub 

I will try to answer your other questions:

**1: How do you publish the events?**
I publish the event in a method that is part of a controller in a Web API Project (Pub). I used "send" method from Rebus.NET

![image](https://github.com/rebus-org/Rebus/assets/5134473/4662fa13-2855-4c07-89ab-fd7fe3bb6f1f)

**2: Where do you subscribe to them?**
I subscribe the events in a console project that always runs. In this project, I created a Handler that receive the messages.

**3: How do you determine whether an event gets handled or not?**
With a simple test in debug, I put a breakpoint in the method "Handle" in the Event Handler class and wait to arrive the message :)

![image](https://github.com/rebus-org/Rebus/assets/5134473/8e9bdcb7-9f66-4e05-b269-50e1b0f52303)

I attach my example project
[WebApplication1.zip](https://github.com/rebus-org/Rebus/files/14398924/WebApplication1.zip)

Thank you! Greetings!

**Comment by mookid8000** [comment]
Hi @alsahe26 , 

First off, using in-mem transport requires that your sender and your receiver both are running in the same process, as they will need to share the `InMemNetwork` instance to be able to communicate with each other. 

In other words, you need to use a REAL transport (e.g. like RabbitMQ and Azure Service Bus, which you mention) to communicate from your web tier to your background service.

Why it works for you with RabbitMQ and not with Azure Service Bus, I cannot tell from the code you posted. I bet your logs will reveal why 🙂 Could you try looking at them?

**Reply by alsahe26** [reply]
Hi @mookid8000!

Thank you for your suggestion, I will review logs with Azure Service Bus like transportation method.
